pkgs/development/interpreters/python/catch_conflicts/catch_conflicts.py

```diff
@@ -1,17 +1,24 @@
-import pkg_resources
+from importlib.metadata import PathDistribution
+from pathlib import Path
 import collections
 import sys
 
+
 do_abort = False
 packages = collections.defaultdict(list)
 
-for f in sys.path:
-    for req in pkg_resources.find_distributions(f):
-        if req not in packages[req.project_name]:
-            # some exceptions inside buildPythonPackage
-            if req.project_name in ['setuptools', 'pip', 'wheel']:
-                continue
-            packages[req.project_name].append(req)
+
+for path in sys.path:
+    for dist_info in Path(path).glob("*.dist-info"):
+        dist = PathDistribution(dist_info)
+
+        # some exceptions inside buildPythonPackage
+        if dist.name in ("setuptools", "pip", "wheel"):
+            continue
```

Comment:
Not quite sure why we allow these exceptions. Possibly because they were part of the previous bootstrap chain? In that case we could drop that condition entirely.

Reply:
They are probably no longer needed.
I tried to build the lxml and did not see any of the errors mentioned in [the comment.](https://github.com/NixOS/nixpkgs/pull/11086#issuecomment-157988574)

Reply:
Removed.

Reply:
> Not quite sure why we allow these exceptions. Possibly because they were part of the previous bootstrap chain? In that case we could drop that condition entirely.

That's correct.